### PR TITLE
Feature 4 editing flashcard set

### PR DESCRIPTION
**Pull request by ThomasShimUoft**
Implemented edit Flashcards and edit FlashcardSets including GUI.

### CODE REVIEW

**Review by Edward-1117 — COMMENTED** [review]
Looks pretty good! I like the file naming conventions you used to reflect the Clean Architecture model. Only thing I think is missing is the test case for FCSetEditorInteractor.

File: src/main/java/Editor/FlashcardSet/FCSetEditorInteractor.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
+package Editor.FlashcardSet;
+
+import dataAccess.DBGateway;
+import dataAccess.IFlashcardSetDataAccess;
+import entityRequestModels.FlashcardSetDsRequestModel;
+
+import java.util.List;
+
+public class FCSetEditorInteractor implements FCSetEditorInputBoundary {
```
[the remark this answers is not in the document]
Comment (ThomasShimUoft) [reply src/main/java/Editor/FlashcardSet/FCSetEditorInteractor.java:9]: Yes thank you for the advice, I'll add test cases for FCSetEditor as well.

**Review by Junyu11111 — APPROVED** [review]
All good

**Comment by ThomasShimUoft** [comment]
This implemented use case adheres to Clean Architecture and SOLID Principles. The code follows exactly each of the clean architecture layers and maintains the boundaries needed to make it clean. The single responsibility principle is shown as every class has a role whether it be an interactor or a presenter and so each class has one responsibility, this follows directly from clean architecture. The open/closed principle is shown as if we ever want to edit another special type of flashcard we can simply make new input/output data, and input/output boundaries as the code do not have hard code dependencies, so no modification but extension. Liskov substitution principle is shown with the interfaces such as the boundaries. The interactor implements the input boundary and since the input boundary has no implementations we can use the input boundary as our type for interactors, this also helps us adhere to clean architecture using dependency injection. Interface segregation principle, in my boundary classes there are only the needed methods so that the interactor and the presenters do not depend on things they do not need. Dependency inversion principle, we see that by following clean architecture we use dependency inversion to allow us to use classes from lower-level classes to upper-level classes, such as the boundary classes.

**Pull request merged**